pkg/telemetry/providers/prometheus/prometheus.go

```diff
@@ -0,0 +1,51 @@
+// Package prometheus provides Prometheus metric exporter implementation
+package prometheus
+
+import (
+	"fmt"
+	"net/http"
+
+	promclient "github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/collectors"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"go.opentelemetry.io/otel/exporters/prometheus"
+	sdkmetric "go.opentelemetry.io/otel/sdk/metric"
+)
+
+// Config holds Prometheus-specific configuration
+type Config struct {
+	// EnableMetricsPath controls whether to expose Prometheus-style /metrics endpoint
+	EnableMetricsPath bool
+	// IncludeRuntimeMetrics adds Go runtime metrics to the registry
+	IncludeRuntimeMetrics bool
+}
+
+// NewReader creates a Prometheus metric reader and HTTP handler for use in a unified meter provider
+func NewReader(config Config) (sdkmetric.Reader, http.Handler, error) {
+	if !config.EnableMetricsPath {
+		return nil, nil, fmt.Errorf("prometheus provider requires EnableMetricsPath to be true")
+	}
+
+	// Create a dedicated registry
+	registry := promclient.NewRegistry()
+
+	// Add runtime metrics if requested
+	if config.IncludeRuntimeMetrics {
+		registry.MustRegister(collectors.NewGoCollector())
```

Comment:
is there any other kind of collector needed? i saw "BuildCollector", but i guess it's not important for now... but wanted to just raise the point

Reply:
Yeah, we could always add `collectors.NewBuildInfoCollector()` in future if we wanted/needed.